src/openforms/submissions/models/submission.py

```diff
@@ -708,7 +708,13 @@ def calculate_price(self, save=True) -> None:
 
     @property
     def payment_required(self) -> bool:
-        return self.form.payment_required
+        if not self.form.payment_required:
+            return False
+
+        self.calculate_price()
+        if self.price:
+            return True
+        return False
```

Comment:
```suggestion
        self.calculate_price(save=False)
        return bool(self.price)
```

Last line is just a personal preference, however I think we shouldn't save on a property access.